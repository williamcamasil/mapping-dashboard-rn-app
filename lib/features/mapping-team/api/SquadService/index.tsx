export type CollaboratorsPropsType = {
  Id: number,
  Name: string,
  Office: string,
  Project: string,
  Squad: string,
  Company: string,
  Hired: number,
  Skills: string,
  Email: string,
  Location: string,
  Responsibilities: string,
};

export type SquadPropsType = {
  Id: number,
  Name: string,
  Description: string,
  Manager: string,
  Collaborators: CollaboratorsPropsType[];
};

export type ChartScreenPropsType = {
  team: string;
  squads: SquadPropsType[];
};

export const squadList = {
  team: 'Omni Bulls',
  squads: [
    {
      Id: 1,
      Name: 'Arquitetura',
      Description: 'Manter a arquitetura do projeto funcionando, performando e com qualidade.',
      Manager: 'Yano',
      Collaborators: [
        {
          Id: 1,
          Name: 'Adriano Mendes',
          Office: 'Gestor',
          Project: 'Omni Bulls',
          Squad: 'Arquitetura',
          Company: 'DB1',
          Hired: 2018,
          Skills: 'Gerir projetos e pessoas',
          Email: 'adriano.mendes@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Gestão de pessoas e projeto',
        },
        {
          Id: 2,
          Name: 'Eloi Rodrigues',
          Office: 'Líder Técnico',
          Project: 'Omni Bulls',
          Squad: 'Arquitetura',
          Company: 'Omni',
          Hired: 2013,
          Skills: 'Gerir squad de arquitetura',
          Email: 'eloi.rodrigues@omni.com.br',
          Location: 'São Paulo (Campinas)',
          Responsibilities: 'Manter a arquitetura do projeto com qualidade e em evolução',
        },
        {
          Id: 3,
          Name: 'Lucas Magalhães',
          Office: 'Tech Leader',
          Project: 'Omni Bulls',
          Squad: 'Arquitetura',
          Company: 'DB1',
          Hired: 2015,
          Skills: 'Apoiar nos desafios técnicos e orientar evolução de carreira e habilidades',
          Email: 'lucas.magalhaes@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Apoio técnico',
        },
        {
          Id: 4,
          Name: 'Douglas Silva',
          Office: 'Desenvolvedor FrontEnd',
          Project: 'Omni Bulls',
          Squad: 'Arquitetura',
          Company: 'DB1',
          Hired: 2008,
          Skills: 'React Native, React Js, Angular, Jest, Java Script, Type Script, ...',
          Email: 'douglas.silva@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Manter a arquitetura do projeto, desenvolver novas features e dar manutenção no sistema',
        },
        {
          Id: 5,
          Name: 'Fernanda Sonata',
          Office: 'Desenvolvedor FrontEnd',
          Project: 'Omni Bulls',
          Squad: 'Arquitetura',
          Company: 'DB1',
          Hired: 2014,
          Skills: 'React Native, React Js, Angular, Jest, Java Script, Type Script, ...',
          Email: 'fernanda.sonata@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Manter a arquitetura do projeto, desenvolver novas features e dar manutenção no sistema',
        },
        {
          Id: 6,
          Name: 'William Machado',
          Office: 'Desenvolvedor FrontEnd',
          Project: 'Omni Bulls',
          Squad: 'Arquitetura',
          Company: 'DB1',
          Hired: 2021,
          Skills: 'React Native, React Js, Angular, Jest, Java Script, Type Script, ...',
          Email: 'william.machado@db1.com.br',
          Location: 'São Paulo (Capital)',
          Responsibilities: 'Manter a arquitetura do projeto, desenvolver novas features e dar manutenção no sistema',
        },
        {
          Id: 7,
          Name: 'Júlia Yto',
          Office: 'Quality Assurance',
          Project: 'Omni Bulls',
          Squad: 'Arquitetura',
          Company: 'DB1',
          Hired: 2017,
          Skills: 'Testes de fluxos e serviços',
          Email: 'julia.yto@db1.com.br',
          Location: 'Santa Catarina (Capital)',
          Responsibilities: 'Manter a arquitetura do projeto, desenvolver testes manter o sistema',
        },
      ]
    },
    {
      Id: 2,
      Name: 'Segurança',
      Description: 'Teste 1 Lorem Ipsum é simplesmente uma simulação de texto da indústria tipográfica e de impressos, e vem sendo utilizado desde o século XVI.',
      Manager: 'Teste 2',
      Collaborators: [
        {
          Id: 1,
          Name: 'Teste Silva',
          Office: 'Gestor',
          Project: 'Omni Bulls',
          Squad: 'Segurança',
          Company: 'DB1',
          Hired: 2018,
          Skills: 'Gerir projetos e pessoas',
          Email: 'teste.silva@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Gestão de pessoas e projeto',
        },
        {
          Id: 2,
          Name: 'Teste Vieira',
          Office: 'Líder Técnico',
          Project: 'Omni Bulls',
          Squad: 'Segurança',
          Company: 'Omni',
          Hired: 2013,
          Skills: 'Gerir squad',
          Email: 'teste.vieira@omni.com.br',
          Location: 'São Paulo (Campinas)',
          Responsibilities: 'Manter o projeto com qualidade e em evolução',
        },
        {
          Id: 3,
          Name: 'Teste Santos',
          Office: 'Tech Leader',
          Project: 'Omni Bulls',
          Squad: 'Segurança',
          Company: 'DB1',
          Hired: 2015,
          Skills: 'React Native, React Js, Angular, Jest, Java Script, Type Script, ...',
          Email: 'teste.santos@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Manter o projeto, desenvolver novas features e dar manutenção no sistema',
        },
      ]
    },
    {
      Id: 3,
      Name: 'Regulatório',
      Description: 'Teste 2 Lorem Ipsum é simplesmente uma simulação de texto da indústria tipográfica e de impressos, e vem sendo utilizado desde o século XVI.',
      Manager: 'Teste 3',
      Collaborators: [
        {
          Id: 1,
          Name: 'Teste Silva',
          Office: 'Gestor',
          Project: 'Omni Bulls',
          Squad: 'Regulatório',
          Company: 'DB1',
          Hired: 2018,
          Skills: 'Gerir projetos e pessoas',
          Email: 'teste.silva@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Gestão de pessoas e projeto',
        },
        {
          Id: 2,
          Name: 'Teste Vieira',
          Office: 'Líder Técnico',
          Project: 'Omni Bulls',
          Squad: 'Regulatório',
          Company: 'Omni',
          Hired: 2013,
          Skills: 'Gerir squad de arquitetura',
          Email: 'teste.vieira@omni.com.br',
          Location: 'São Paulo (Campinas)',
          Responsibilities: 'Manter o projeto com qualidade e em evolução',
        },
        {
          Id: 3,
          Name: 'Teste Santos',
          Office: 'Tech Leader',
          Project: 'Omni Bulls',
          Squad: 'Regulatório',
          Company: 'DB1',
          Hired: 2015,
          Skills: 'React Native, React Js, Angular, Jest, Java Script, Type Script, ...',
          Email: 'teste.santos@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Manter o projeto, desenvolver novas features e dar manutenção no sistema',
        },
      ]
    },
    {
      Id: 4,
      Name: 'Operação',
      Description: 'Teste 3 Lorem Ipsum é simplesmente uma simulação de texto da indústria tipográfica e de impressos, e vem sendo utilizado desde o século XVI.',
      Manager: 'Teste 4',
      Collaborators: [
        {
          Id: 1,
          Name: 'Teste Silva',
          Office: 'Gestor',
          Project: 'Omni Bulls',
          Squad: 'Operação',
          Company: 'DB1',
          Hired: 2018,
          Skills: 'Gerir projetos e pessoas',
          Email: 'teste.silva@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Gestão de pessoas e projeto',
        },
        {
          Id: 2,
          Name: 'Teste Vieira',
          Office: 'Líder Técnico',
          Project: 'Omni Bulls',
          Squad: 'Operação',
          Company: 'Omni',
          Hired: 2013,
          Skills: 'Gerir squad de arquitetura',
          Email: 'teste.vieira@omni.com.br',
          Location: 'São Paulo (Campinas)',
          Responsibilities: 'Manter o projeto com qualidade e em evolução',
        },
        {
          Id: 3,
          Name: 'Teste Santos',
          Office: 'Tech Leader',
          Project: 'Omni Bulls',
          Squad: 'Operação',
          Company: 'DB1',
          Hired: 2015,
          Skills: 'React Native, React Js, Angular, Jest, Java Script, Type Script, ...',
          Email: 'teste.santos@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Manter o projeto, desenvolver novas features e dar manutenção no sistema',
        },
      ]
    },
    {
      Id: 5,
      Name: 'Sustentação',
      Description: 'Teste 4 Lorem Ipsum é simplesmente uma simulação de texto da indústria tipográfica e de impressos, e vem sendo utilizado desde o século XVI.',
      Manager: 'Teste 5',
      Collaborators: [
        {
          Id: 1,
          Name: 'Teste Silva',
          Office: 'Gestor',
          Project: 'Omni Bulls',
          Squad: 'Sustentação',
          Company: 'DB1',
          Hired: 2018,
          Skills: 'Gerir projetos e pessoas',
          Email: 'teste.silva@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Gestão de pessoas e projeto',
        },
        {
          Id: 2,
          Name: 'Teste Vieira',
          Office: 'Líder Técnico',
          Project: 'Omni Bulls',
          Squad: 'Sustentação',
          Company: 'Omni',
          Hired: 2013,
          Skills: 'Gerir squad de arquitetura',
          Email: 'teste.vieira@omni.com.br',
          Location: 'São Paulo (Campinas)',
          Responsibilities: 'Manter o projeto com qualidade e em evolução',
        },
        {
          Id: 3,
          Name: 'Teste Santos',
          Office: 'Tech Leader',
          Project: 'Omni Bulls',
          Squad: 'Sustentação',
          Company: 'DB1',
          Hired: 2015,
          Skills: 'React Native, React Js, Angular, Jest, Java Script, Type Script, ...',
          Email: 'teste.santos@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Manter o projeto, desenvolver novas features e dar manutenção no sistema',
        },
      ]
    },
    {
      Id: 6,
      Name: 'Negócios',
      Description: 'Teste 5 Lorem Ipsum é simplesmente uma simulação de texto da indústria tipográfica e de impressos, e vem sendo utilizado desde o século XVI.',
      Manager: 'Teste 6',
      Collaborators: [
        {
          Id: 1,
          Name: 'Teste Silva',
          Office: 'Gestor',
          Project: 'Omni Bulls',
          Squad: 'Negócios',
          Company: 'DB1',
          Hired: 2018,
          Skills: 'Gerir projetos e pessoas',
          Email: 'teste.silva@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Gestão de pessoas e projeto',
        },
        {
          Id: 2,
          Name: 'Teste Vieira',
          Office: 'Líder Técnico',
          Project: 'Omni Bulls',
          Squad: 'Negócios',
          Company: 'Omni',
          Hired: 2013,
          Skills: 'Gerir squad de arquitetura',
          Email: 'teste.vieira@omni.com.br',
          Location: 'São Paulo (Campinas)',
          Responsibilities: 'Manter o projeto com qualidade e em evolução',
        },
        {
          Id: 3,
          Name: 'Teste Santos',
          Office: 'Tech Leader',
          Project: 'Omni Bulls',
          Squad: 'Negócios',
          Company: 'DB1',
          Hired: 2015,
          Skills: 'React Native, React Js, Angular, Jest, Java Script, Type Script, ...',
          Email: 'teste.santos@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Manter o projeto, desenvolver novas features e dar manutenção no sistema',
        },
      ]
    },
    {
      Id: 7,
      Name: 'Serviço Bancário e Negócios PJ',
      Description: 'Teste 6 Lorem Ipsum é simplesmente uma simulação de texto da indústria tipográfica e de impressos, e vem sendo utilizado desde o século XVI.',
      Manager: 'Teste 7',
      Collaborators: [
        {
          Id: 1,
          Name: 'Teste Silva',
          Office: 'Gestor',
          Project: 'Omni Bulls',
          Squad: 'Serviço Bancário e Negócios PJ',
          Company: 'DB1',
          Hired: 2018,
          Skills: 'Gerir projetos e pessoas',
          Email: 'teste.silva@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Gestão de pessoas e projeto',
        },
        {
          Id: 2,
          Name: 'Teste Vieira',
          Office: 'Líder Técnico',
          Project: 'Omni Bulls',
          Squad: 'Serviço Bancário e Negócios PJ',
          Company: 'Omni',
          Hired: 2013,
          Skills: 'Gerir squad de arquitetura',
          Email: 'teste.vieira@omni.com.br',
          Location: 'São Paulo (Campinas)',
          Responsibilities: 'Manter o projeto com qualidade e em evolução',
        },
        {
          Id: 3,
          Name: 'Teste Santos',
          Office: 'Tech Leader',
          Project: 'Omni Bulls',
          Squad: 'Serviço Bancário e Negócios PJ',
          Company: 'DB1',
          Hired: 2015,
          Skills: 'React Native, React Js, Angular, Jest, Java Script, Type Script, ...',
          Email: 'teste.santos@db1.com.br',
          Location: 'Paraná (Maringá)',
          Responsibilities: 'Manter o projeto, desenvolver novas features e dar manutenção no sistema',
        },
      ]
    },
  ]
};
