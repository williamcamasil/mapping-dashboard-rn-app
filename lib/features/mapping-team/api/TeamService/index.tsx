import { CollaboratorsPropsType } from "../SquadService";

export enum TeamStructureEnum {
  EQUIPE = 'EQUIPE',
  ORGANOGRAMA = 'ORGANOGRAMA',
};

export enum AccessTypeEnum {
  ADMIN = 'ADMIN',
  NORMAL = 'NORMAL',
};

export type MemberPropsType = {
  Id: number,
  Main: boolean,
  Structure: string,
  Name: string,
  Type: string,
  Collaborator?: CollaboratorsPropsType;
};

export type OverviewScreenPropsType = {
  team: string;
  userType: string;
  members: MemberPropsType[];
};

export type FormType = 'create' | 'edit';

export const teamList = {
  team: 'Omni Bulls',
  userType: AccessTypeEnum.ADMIN,
  members: [
    {
      Id: 1,
      Main: true,
      Structure: TeamStructureEnum.EQUIPE,
      Name: 'Edi Matias',
      Type: 'CO do projeto',
      Collaborator: {
        Id: 1,
        Name: 'Edi Matias',
        Office: 'CO do projeto',
        Project: 'Omni Bulls',
        Squad: '',
        Company: 'Omni',
        Hired: 1997,
        Skills: 'Gerir times',
        Email: 'edi.matias@omni.com.br',
        Location: 'São Paulo (Capital)',
        Responsibilities: 'Gestão de projetos',
      }
    },
    {
      Id: 2,
      Main: false,
      Structure: TeamStructureEnum.EQUIPE,
      Name: 'Sandra',
      Type: 'Gerente de portfólio',
      Collaborator: {
        Id: 1,
        Name: 'Sandra',
        Office: 'Gerente de portfólio',
        Project: 'Omni Bulls',
        Squad: '',
        Company: 'DB1',
        Hired: 1997,
        Skills: 'Gerir times',
        Email: 'sandra.larezzi@db1.com.br',
        Location: 'Ceará (Capital)',
        Responsibilities: 'Gestão de projetos',
      }
    },
    {
      Id: 3,
      Main: false,
      Structure: TeamStructureEnum.ORGANOGRAMA,
      Name: 'Organograma das squads',
      Type: 'Omni Bulls',
    },
  ]
};
