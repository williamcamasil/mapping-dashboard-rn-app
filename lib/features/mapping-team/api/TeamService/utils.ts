import { CollaboratorsPropsType } from "../SquadService/utils";

export enum TeamStructureEnum {
  EQUIPE = 'EQUIPE',
  ORGANOGRAMA = 'ORGANOGRAMA',
};

export enum AccessTypeEnum {
  ADMIN = 'ADMIN',
  NORMAL = 'NORMAL',
};

export type MemberPropsType = {
  Id: number,
  Main: boolean,
  Structure: string,
  Name: string,
  Type: string,
  Collaborator?: CollaboratorsPropsType;
};

export type OverviewScreenPropsType = {
  team: string;
  userType: string;
  members: MemberPropsType[];
};

export type FormType = 'create' | 'edit';
