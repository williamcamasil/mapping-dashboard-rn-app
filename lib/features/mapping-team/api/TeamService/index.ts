import { AccessTypeEnum, TeamStructureEnum } from "./utils";

export const serviceTeamList = {
  team: 'Omni Bulls',
  userType: AccessTypeEnum.ADMIN,
  members: [
    {
      Id: 1,
      Main: true,
      Structure: TeamStructureEnum.EQUIPE,
      Name: 'Edi Matias',
      Type: 'CO do projeto',
      Collaborator: {
        Id: 1,
        Name: 'Edi Matias',
        Office: 'CO do projeto',
        Project: 'Omni Bulls',
        Squad: '',
        Company: 'Omni',
        Hired: 1997,
        Skills: 'Gerir times',
        Email: 'edi.matias@omni.com.br',
        Location: 'São Paulo (Capital)',
        Responsibilities: 'Gestão de projetos',
      }
    },
    {
      Id: 2,
      Main: false,
      Structure: TeamStructureEnum.EQUIPE,
      Name: 'Sandra',
      Type: 'Gerente de portfólio',
      Collaborator: {
        Id: 1,
        Name: 'Sandra',
        Office: 'Gerente de portfólio',
        Project: 'Omni Bulls',
        Squad: '',
        Company: 'DB1',
        Hired: 1997,
        Skills: 'Gerir times',
        Email: 'sandra.larezzi@db1.com.br',
        Location: 'Ceará (Capital)',
        Responsibilities: 'Gestão de projetos',
      }
    },
    {
      Id: 3,
      Main: false,
      Structure: TeamStructureEnum.ORGANOGRAMA,
      Name: 'Organograma das squads',
      Type: 'Omni Bulls',
    },
  ]
};
