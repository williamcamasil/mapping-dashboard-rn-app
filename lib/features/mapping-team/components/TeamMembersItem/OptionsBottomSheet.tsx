import React, { useCallback } from 'react';

import {
  AlertModal,
  BottomSheet, BottomSheetPropsType, Container, MenuOption, Spacer, Text, useDismissAllModals, useModal, useTheme,
} from 'mapping-style-guide-rn';
import { MemberPropsType, TeamStructureEnum } from '../../api/TeamService';
import { getNavigationHolder } from 'mapping-context-rn';

type OptionsBottomSheetType = BottomSheetPropsType & {
  squadData: MemberPropsType;
  onItemDeleted?: () => void;
};

const OptionsBottomSheet = ({
  squadData, onDismiss, onItemDeleted, ...others
}: OptionsBottomSheetType) => {
  const theme = useTheme();
  const showModal = useModal();
  const dismissAllModals = useDismissAllModals();

  const handleCloseModal = useCallback(() => {
    dismissAllModals();
    onItemDeleted?.();
  }, [dismissAllModals, onItemDeleted]);

  const handleDeleteItem = useCallback(() => {
    showModal(AlertModal, {
      title: 'Item excluído com sucesso!',
      description: 'Item excluído com sucesso. Esse item não será mostrado na lista.',
      primaryButtonName: 'Ok',
      onPressPrimary: handleCloseModal,
    });
    onDismiss?.();
  }, [squadData, onDismiss, showModal, onItemDeleted]);

  const handleEditItemList = useCallback(() => {
    getNavigationHolder().navigate('FormOverviewScreen', { editedItem: squadData, formType: 'edit' });
    onDismiss?.();
  }, [squadData,  onDismiss, showModal]);

  const handleEditItemListInformation = useCallback(() => {
    getNavigationHolder().navigate('FormSquadScreen', { editedItem: squadData.Collaborator, formType: 'edit' });
    onDismiss?.();
  }, [squadData,  onDismiss, showModal]);

  return (
    <BottomSheet {...others} onDismiss={onDismiss} testID='team-members-bottom-sheet'>
      <BottomSheet.Title>O que deseja fazer?</BottomSheet.Title>
      <Text>Escolha uma opção</Text>
      <Spacer size={theme.spacings.sLarge} />
      <Container>
        <MenuOption onPress={handleEditItemList} description="Edite as informações">
          Editar item da lista
        </MenuOption>
        {squadData.Structure !== TeamStructureEnum.EQUIPE ? null : (
          <MenuOption onPress={handleEditItemListInformation} description="Edite informações da pessoa">
            Editar informações da pessoa
          </MenuOption>
        )}
        <MenuOption onPress={handleDeleteItem} description="Retire item da sua lista">
          Excluir item da lista
        </MenuOption>
      </Container>
    </BottomSheet>
  );
};

export default OptionsBottomSheet;
