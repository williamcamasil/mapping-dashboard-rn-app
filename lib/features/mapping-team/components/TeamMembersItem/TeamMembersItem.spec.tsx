import React from 'react';

import {
  render, waitFor, fireEvent, waitForElementToBeRemoved,
} from '@testing-library/react-native';
import { ModalProvider } from 'mapping-style-guide-rn';

import TeamMembersItem from '.';
import { MemberPropsType, TeamStructureEnum } from '../../api/TeamService';

const defaultItemDataMock = {
  Id: 1,
  Main: true,
  Structure: TeamStructureEnum.EQUIPE,
  Name: 'Edi Matias',
  Type: 'CO do projeto',
  Collaborator: {
    Id: 1,
    Name: 'Edi Matias',
    Office: 'CO do projeto',
    Project: 'Omni Bulls',
    Squad: '',
    Company: 'Omni',
    Hired: 1997,
    Skills: 'Gerir times',
    Email: 'edi.matias@omni.com.br',
    Location: 'São Paulo (Capital)',
    Responsibilities: 'Gestão de projetos',
  },
};

const itemDataMockNoStar = {
  ...defaultItemDataMock,
  Main: false,
  Structure: TeamStructureEnum.ORGANOGRAMA,
  Name: 'Matias Federico',
};

const renderComponent = async (
  itemData: MemberPropsType = defaultItemDataMock,
  onItemPress: () => void = jest.fn(),
  onItemDeleted: () => void = jest.fn(),
  isAdminUser: boolean = true,
) => {
  const component = render(
    <ModalProvider>
      <TeamMembersItem
        itemData={itemData}
        onItemPress={onItemPress}
        onItemDeleted={onItemDeleted}
        isAdminUser={isAdminUser}
      />
    </ModalProvider>,
  );
  await waitFor(() => component.UNSAFE_getByType(TeamMembersItem));
  return component;
};

describe('TeamMembersItem', () => {
  it('Should match snapshot', async () => {
    const component = await renderComponent();
    expect(component).toMatchSnapshot();
  });

  it('Should render the component with admin prop as star item', async () => {
    const component = await renderComponent();
    component.getByText('Edi Matias');
    component.getByTestId('icon-dots');
    component.getByTestId('icon-star');
  });

  it('Should click on item press', async () => {
    const onItemPress = jest.fn();
    const component = await renderComponent(defaultItemDataMock, onItemPress);
    component.getByText('Edi Matias');
    fireEvent.press(component.getByTestId('item-list'));
    expect(onItemPress).toBeCalled();
  });

  it('Should render the component item as not star item', async () => {
    const component = await renderComponent(itemDataMockNoStar);
    component.getByText('Matias Federico');
    component.queryByTestId('icon-star');
  });

  it('Should render the component without admin prop', async () => {
    const component = await renderComponent(defaultItemDataMock, jest.fn(), jest.fn(), false);
    component.getByText('Edi Matias');
    component.queryByTestId('icon-dots');
  });

  it('Should click on icon dots, show bottom sheet with options', async () => {
    const component = await renderComponent();
    component.getByText('Edi Matias');
    fireEvent.press(component.getByTestId('icon-dots'));
    await component.findByTestId('team-members-bottom-sheet');
    component.getByText('Editar item da lista');
    component.getByText('Editar informações da pessoa');
    component.getByText('Excluir item da lista');
  });

  it('Should click on icon dots, show bottom sheet with options without edit personal information option', async () => {
    const component = await renderComponent(itemDataMockNoStar);
    component.getByText('Matias Federico');
    fireEvent.press(component.getByTestId('icon-dots'));
    await component.findByTestId('team-members-bottom-sheet');
    component.getByText('Editar item da lista');
    component.queryByTestId('Editar informações da pessoa');
    component.getByText('Excluir item da lista');
  });

  it('Should click on icon dots, show bottom sheet with options, click on delete item and close modal', async () => {
    const component = await renderComponent(itemDataMockNoStar);
    component.getByText('Matias Federico');
    fireEvent.press(component.getByTestId('icon-dots'));
    await component.findByTestId('team-members-bottom-sheet');
    fireEvent.press(component.getByText('Excluir item da lista'));
    await component.findByText('Item excluído com sucesso!');
    fireEvent.press(component.getByText('Ok'));
    await waitForElementToBeRemoved(() => component.queryByText('Item excluído com sucesso!'));
  });
});
