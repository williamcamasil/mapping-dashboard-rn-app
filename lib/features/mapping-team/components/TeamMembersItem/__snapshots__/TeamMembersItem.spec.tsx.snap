// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamMembersItem Should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingRight": 12,
    }
  }
  testID="item-list"
>
  <View
    style={
      {
        "flex": 1,
        "maxWidth": "88%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        allowFontScaling={false}
        numberOfLines={1}
        style={
          {
            "color": "#3E4149",
            "flexShrink": 1,
            "fontFamily": "Averta CY",
            "fontSize": 16,
            "fontWeight": "bold",
            "letterSpacing": -0.02,
            "lineHeight": 22,
          }
        }
      >
        Edi Matias
      </Text>
      <Text
        style={
          {
            "height": 8,
            "width": 8,
          }
        }
      />
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        fill="none"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        testID="icon-star"
        vbHeight={20}
        vbWidth={22}
        width={16}
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M6.45811 18.8596C5.87407 19.1729 5.14668 18.9534 4.83341 18.3694C4.70711 18.1339 4.66301 17.863 4.7081 17.5996L5.5761 12.5766L1.9001 9.01765C1.42757 8.55295 1.42122 7.79318 1.88591 7.32065C2.06948 7.13399 2.30905 7.01248 2.5681 6.97465L7.6501 6.24165L9.9201 1.66765C10.2192 1.07007 10.9462 0.828147 11.5437 1.12729C11.7775 1.24431 11.9671 1.43389 12.0841 1.66765L14.3541 6.24165L19.4361 6.97465C20.0919 7.07041 20.5459 7.67967 20.4501 8.33546C20.4123 8.59451 20.2908 8.83408 20.1041 9.01765L16.4261 12.5746L17.2941 17.5976C17.4072 18.2507 16.9694 18.8717 16.3164 18.9848C16.0522 19.0305 15.7803 18.9865 15.5441 18.8596L11.0001 16.4906L6.45811 18.8596Z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
            propList={
              [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4284910024,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="1.8"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        {
          "height": 4,
          "width": 4,
        }
      }
    />
    <Text
      allowFontScaling={false}
      style={
        {
          "color": "#676D79",
          "flexShrink": 1,
          "fontFamily": "Averta CY",
          "fontSize": 16,
          "fontWeight": "normal",
          "letterSpacing": -0.02,
          "lineHeight": 22,
        }
      }
    >
      CO do projeto
    </Text>
  </View>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="24"
    bbWidth="24"
    focusable={false}
    height={24}
    meetOrSlice={0}
    minX={0}
    minY={0}
    onPress={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    responsible={true}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    testID="icon-dots"
    vbHeight={32}
    vbWidth={32}
    width={24}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    >
      <RNSVGGroup
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        name="Layer_2"
      >
        <RNSVGPath
          d="M16 7a2 2 0 112-2 2 2 0 01-2 2zm0-2zm0 0zm0 0zm0 0zm0 0zm0 0zm0 0zm0 0zM16 18a2 2 0 112-2 2 2 0 01-2 2zm0-2zm0 0zm0 0zm0 0zm0 0zm0 0zm0 0zm0 0zM16 29a2 2 0 112-2 2 2 0 01-2 2zm0-2zm0 0zm0 0zm0 0zm0 0zm0 0zm0 0zm0 0z"
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          propList={
            [
              "strokeWidth",
            ]
          }
          strokeWidth="1.8"
        />
      </RNSVGGroup>
      <RNSVGPath
        d="M0 0H32V32H0z"
        fill={null}
        name="frame"
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
